R/pkg/R/functions.R

```diff
@@ -3348,18 +3348,14 @@ unresolved_named_lambda_var <- function(...) {
 #'
 #' @param fun R \code{function} (unary, binary or ternary)
 #'        that transforms \code{Columns} into a \code{Column}
-#' @param expected_nargs numeric a vector of the expected
-#'        number of arguments. Used for validation
 #' @return JVM \code{LambdaFunction} object
-create_lambda <- function(fun, expected_nargs) {
+create_lambda <- function(fun) {
   as_jexpr <- function(x) callJMethod(x@jc, "expr")
 
   # Process function arguments
   parameters <- formals(fun)
   nparameters <- length(parameters)
 
-  stopifnot(nparameters %in% expected_nargs)
```

Comment:
I think we might just have to add this line to each function implementation later .. but yeah, let's do it separately after this PR to get the things in we're all good first.